File: backend/internal/domain/repository/sales_activity_repository.go
package repository

import (
	"context"
	"crm_wowin_backend/internal/domain/models"
	"time"

	"github.com/google/uuid"
)

type SalesActivityFilter struct {
	SalesID    *uuid.UUID
	LeadID     *uuid.UUID
	CustomerID *uuid.UUID
	DealID     *uuid.UUID
	StartDate  *time.Time
	EndDate    *time.Time
}

type SalesActivityRepository interface {
	Create(ctx context.Context, activity *models.SalesActivity) error
	GetByID(ctx context.Context, id uuid.UUID) (*models.SalesActivity, error)
	List(ctx context.Context, filter SalesActivityFilter) ([]*models.SalesActivity, error)
	Update(ctx context.Context, activity *models.SalesActivity) error
	Delete(ctx context.Context, id uuid.UUID) error
}

File: mobile/lib/features/activities/domain/repositories/sales_activity_repository.dart
import 'package:dartz/dartz.dart';
import '../../../../core/error/failures.dart';
import '../entities/sales_activity.dart';

abstract class SalesActivityRepository {
  Future<Either<Failure, List<SalesActivity>>> getActivities({
    String? salesId,
    String? leadId,
    String? customerId,
    String? dealId,
    DateTime? startDate,
    DateTime? endDate,
  });

  Future<Either<Failure, SalesActivity>> createActivity(SalesActivity activity);
  
  Future<Either<Failure, SalesActivity>> updateActivity(SalesActivity activity);
  
  Future<Either<Failure, Unit>> deleteActivity(String id);
}
